### ISSUE TITLE
A way to organize downloads

### ISSUE THREAD

**Issue by tabishm0n** [get-got/discord-downloader-go#140]
Is there a way to download media into a specific folder if a message contains a specific role that was pinged with it or a specific message/tag?

**Comment by kauffy** [comment]
If you're using a bot on your own server on Discord, then create multiple channels on your server and invite your bot (and the Midjourney bot) to the channel. You'd have to add the channels to your configuration file for DDG, but when the bot downloads, it will put them into folders named for the channel, should you turn on those settings.

**Comment by get-got** [comment]
> Is there a way to download media into a specific folder if a message contains a specific role that was pinged with it or a specific message/tag?

@tabishm0n Yes but not simultaneously with a separate config for that same Discord source.

You can use "filters" -> "allowedPhrases" to only listen for messages containing pings of that role; Example:

```json

"filters": {
	"allowedPhrases": [ "<@1230000000000>" ]
}
```

The way the bot is currently structured, it only binds to the first matched settings group it can find, not multiple.

So you could have a settings group that scrapes all media or you could have a settings group that only scrapes media when this role is pinged, but not both simultaneously, bound to the same Discord source.

I'll keep this in mind for restructuring in the future, to allow for multiple settings groups to apply to the same source, but I'd have to change quite a few integral things so it would be a task, not sure when I'll get time.

**Comment by kauffy** [comment]
The dirty method would simply be to swap configuration files and and restart the application. You might have to use a pass-under if the settings file is held open (name the desired config file *.new, then on restart, look to see if there is a *.new file and rename it and then load it-- no *.new, load the last one).
Not elegant, but it achieves the effect.
